src/Promitor.Agents.Scraper/Validation/MetricDefinitions/ResourceTypes/NatGatewayMetricValidator.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections.Generic;
+using System.Linq;
+using GuardNet;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Promitor.Agents.Scraper.Validation.MetricDefinitions.Interfaces;
+using Promitor.Core.Contracts.ResourceTypes;
+
+namespace Promitor.Agents.Scraper.Validation.MetricDefinitions.ResourceTypes
+{
+    internal class NatGatewayMetricValidator : IMetricValidator
+    {
+        public IEnumerable<string> Validate(MetricDefinition metricDefinition)
+        {
+            Guard.NotNull(metricDefinition, nameof(metricDefinition));
+
+            foreach (var resourceDefinition in metricDefinition.Resources.Cast<NatGatewayResourceDefinition>())
+            {
+                if (string.IsNullOrWhiteSpace(resourceDefinition.NatGatewayName))
+                {
+                    yield return "No Azure NAT Gateway name is configured";
```

Comment:
Nit - Let's align with product naming
```suggestion
                    yield return "No Azure Virtual Network NAT Gateway name is configured";
```

Reply:
Added proper product name